src/map/status.cpp

```diff
@@ -10173,28 +10212,81 @@ int32 status_change_start(struct block_list* src, struct block_list* bl,enum sc_
 
 	int32 tick = (int32)duration;
 
-	sd = BL_CAST(BL_PC, bl);
-	vd = status_get_viewdata(bl);
-
-	undead_flag = battle_check_undead(status->race,status->def_ele);
-	// Check for immunities / sc fails
+	// Type-specific checks that need to happen before the delay
 	switch (type) {
-		case SC_VACUUM_EXTREME:
-			if (sc && sc->getSCE(SC_VACUUM_EXTREME_POSTDELAY) && sc->getSCE(SC_VACUUM_EXTREME_POSTDELAY)->val2 == val2) // Ignore post delay from other vacuum (this will make stack effect enabled)
-				return 0;
-			break;
 		case SC_STONE:
 		case SC_STONEWAIT:
 		case SC_FREEZE:
 			// Undead are immune to Freeze/Stone
-			if (undead_flag && !(flag&SCSTART_NOAVOID))
+			if (battle_check_undead(status->race, status->def_ele) != 0 && !(flag&SCSTART_NOAVOID))
 				return 0;
+			else if (type == SC_STONEWAIT) {
+				// Stonewait has a unique handling where the delay is actually duration until stone kicks in
+				val3 = max(1, tick - delay); // Petrify time
```

Comment:
```suggestion
				val3 = std::max<int32>(1, tick - delay); // Petrify time
```
Probably would need to change it to t_tick eventually but in the meantime...